src/services/shortLink/test/index.test.ts

```diff
@@ -0,0 +1,552 @@
+      expect(response.json()).toEqual({
+        ...payload,
+        url: getRealPath(payload.item_id, payload.platform),
+      });
+    });
+    it('Success when admin', async () => {
+      const { item } = await mockPostData({
+        author: bob,
+        member: actor,
+        permission: PermissionLevel.Admin,
+      });
+      const payload = mockShortLinkPayload({ item_id: item.id, alias: MOCK_ALIAS });
+      const post = await injectPost(app, payload);
+      expect(post.statusCode).toEqual(StatusCodes.OK);
+      expect(post.json().alias).toEqual(MOCK_ALIAS);
+
+      const response = await injectGet(app, MOCK_ALIAS);
+      expect(response.statusCode).toEqual(StatusCodes.OK);
+      expect(response.json()).toEqual({
+        ...payload,
+        url: getRealPath(payload.item_id, payload.platform),
+      });
+    });
+  });
+
+  describe('PATCH /short-links/:itemId', () => {
+    beforeEach(async () => {
+      ({ app, actor } = await build());
+      bob = await saveMember(BOB);
+    });
+
+    it('Not found if patch short links with invalid item id', async () => {
+      const response = await injectPatch(app, MOCK_ALIAS, { platform: GraaspPlatform.BUILDER });
+      expect(response.statusCode).toEqual(StatusCodes.NOT_FOUND);
+    });
+    describe('Unauthorized and forbidden', () => {
+      it('Unauthorized if patch short links without connected member', async () => {
+        const { item } = await mockPostData({ author: actor });
+        const post = await injectPost(app, mockShortLinkPayload({ item_id: item.id }));
+        expect(post.statusCode).toEqual(StatusCodes.OK);
+
+        logOut(app);
+
+        const response = await injectPatch(app, MOCK_ALIAS, { platform: GraaspPlatform.BUILDER });
+        expect(response.statusCode).toEqual(StatusCodes.UNAUTHORIZED);
+      });
+      it('Forbidden if patch short links with write permission', async () => {
+        const { item } = await mockPostData({
+          author: actor,
+          member: bob,
+          permission: PermissionLevel.Write,
+        });
+        const post = await injectPost(app, mockShortLinkPayload({ item_id: item.id }));
+        expect(post.statusCode).toEqual(StatusCodes.OK);
+
+        // sign in as bob
+        ({ app } = await build({ member: bob }));
+
+        const response = await injectPatch(app, MOCK_ALIAS, { platform: GraaspPlatform.BUILDER });
+        expect(response.statusCode).toEqual(StatusCodes.FORBIDDEN);
+      });
+    });
+
+    describe('Success', () => {
+      it('Success if patch short links with admin permission', async () => {
+        const { item } = await mockPostData({
+          author: actor,
+          member: bob,
+          permission: PermissionLevel.Admin,
+        });
+        const post = await injectPost(app, mockShortLinkPayload({ item_id: item.id }));
+        expect(post.statusCode).toEqual(StatusCodes.OK);
+
+        // sign in as bob
+        ({ app } = await build({ member: bob }));
```

Comment:
I dont think you should re-build the app to login as bob ... maybe adding a spy on the attempt authentication would be enough.

Reply:
is this resolved? The code is still there.